### PR TITLE
[Data Table] [Icons] Replace `SortAscendingMajor` and `SortDescendingMajor` / Use them in Data Table

### PR DESCRIPTION
**Pull request by ErickTamayo**
### WHY are these changes introduced?

Fixes https://github.com/Shopify/core-issues/issues/36954 

### WHAT is this pull request doing?

Replaces the old SortAscendingMajor and SortDescendingMajor icons and implements them in the Data Table component

Before: 

image

After: 

image

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

**Comment by mirualves** [comment]
It looks great to me! Is it possible for me to interact with it on a spin link just to be 100% sure?

**Comment by philschoefer** [comment]
@mirualves, you can literally review changes in Storybook actually: https://www.chromatic.com/build?appId=5d559397bae39100201eedc1&number=13131

You can find the link to the above in the "actions" below a PR: https://screenshot.click/08-45-2sdcl-aflgx.png

Here is a short video of the process. This works for any changes in libraries (Polaris Viz, Polaris etc.)
https://user-images.githubusercontent.com/4960217/178044122-427ac0cd-118c-427e-858e-89edbe74d5e9.mp4

File: .changeset/six-seahorses-glow.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+'@shopify/polaris-icons': major
+'@shopify/polaris': major
```
Comment (philschoefer) [inline .changeset/six-seahorses-glow.md]: Sorry, now that I see this, I think for Polaris this is a minor update. For Polaris I'd say a major though. Not sure though in this case. Maybe someone from Polaris can help better discern.

Comment (ErickTamayo) [reply .changeset/six-seahorses-glow.md]: Sounds good. I updated it

**Review by blboyle — APPROVED** [review]
Looks good to me!

**Review by chloerice — APPROVED** [review]
Hey @ErickTamayo this looks great! 🙌🏽  Only thing I'm unsure of is the hover state of the icon, as it makes it hard to see, but that was a regression a while back and not introduced by this PR. Usually the icon should match the interaction state of the button text. @mirualves what do you think?

**Comment by ErickTamayo** [comment]
> Hey @ErickTamayo this looks great! 🙌🏽 Only thing I'm unsure of is the hover state of the icon, as it makes it hard to see, but that was a regression a while back and not introduced by this PR. Usually the icon should match the interaction state of the button text. @mirualves what do you think?

Thanks @chloerice ! The hover state is going to be addressed in another PR as per @philschoefer

**Pull request merged**